tests/test_emission_calculation.py

```diff
@@ -0,0 +1,119 @@
+import datetime
+
+import pytest
+from qgis.core import QgsMapLayer, QgsVectorLayer
+from qgis.testing import start_app
+from qgis.testing.mocked import get_iface
+
+from open_alaqs.core.alaqsdblite import ProjectDatabase
+from open_alaqs.core.modules.ModuleManager import OutputAnalysisModuleRegistry
+from open_alaqs.openalaqsdialog import OpenAlaqsResultsAnalysis
+from tests.utils import get_copy_path, get_data_path
+
+start_app()
+
+
+@pytest.fixture(scope="module")
+def plugin_instance():
+    print("\nINFO: Get plugin instance")
+    from open_alaqs.openalaqs import OpenALAQS
+
+    plugin = OpenALAQS(get_iface())
+    yield plugin
+
+    print(" [INFO] Tearing down plugin instance")
+    plugin.unload()
+
+
+@pytest.fixture(scope="module")
+def datasets_to_test():
+    print("\nINFO: Get datasets to test...")
+    return [
+        {
+            "title": "EHRD (Rotterdam, NL) Emission calculation test (CO)",
+            "db_path": str(get_data_path("EHRD") / "EHRD.alaqs"),
+            "inventory_path": str(get_data_path("EHRD") / "EHRD_out.alaqs"),
+            "module_name": "EmissionsQGISVectorLayerOutputModule",
+            "pollutant": "CO",
+            "study_start_date": datetime.datetime(2025, 12, 1, 6, 0),
+            "study_end_date": datetime.datetime(2025, 12, 1, 7, 0),
+            "vector_layer_path": str(
+                get_copy_path(get_data_path("EHRD") / "vector_layer_co.gpkg")
+            )
+            + "|layername=output",
+        },
+        {
+            "title": "EHRD (Rotterdam, NL) Emission calculation test (PM10)",
+            "db_path": str(get_data_path("EHRD") / "EHRD.alaqs"),
+            "inventory_path": str(get_data_path("EHRD") / "EHRD_out.alaqs"),
+            "module_name": "EmissionsQGISVectorLayerOutputModule",
+            "pollutant": "PM10",
+            "study_start_date": datetime.datetime(2025, 12, 1, 6, 0),
```

Comment:
One suggestioon: for tests it would be easier if you reduce the python code as much as possible, so you define this as ISO string instead (scalar) and not a datetime (object). This will allow you in theory to parametrize the test from separate files.


```suggestion
            "study_start_date": "2025-12-01 06:00:00",
```

Btw, in newer python versions you should avoid naive dt.